drucker/drucker_dashboard_servicer.py

```diff
@@ -149,27 +154,35 @@ def SwitchModel(self,
         return drucker_pb2.ModelResponse(status=1,
                                          message='Success: Switching model file.')
 
+    @error_handling(drucker_pb2.EvaluateModelResponse(
+        metrics=drucker_pb2.EvaluationMetrics(num=0, accuracy=0.0, precision=[0.0], recall=[0.0], fvalue=[0.0], option={})))
     def EvaluateModel(self,
                       request_iterator: Iterator[drucker_pb2.EvaluateModelRequest],
                       context: _Context
                       ) -> drucker_pb2.EvaluateModelResponse:
-        """ Evaluate your ML model.
-        :TODO: in detail.
+        """ Evaluate your ML model and save result.
         """
-        try:
-            for evaluateModelRequest in request_iterator:
-                test_data = evaluateModelRequest.data
-                result = self.app.evaluate(test_data)
-                return drucker_pb2.EvaluateModelResponse(num=result.num,
-                                                         accuracy=result.accuracy,
-                                                         precision=result.precision,
-                                                         recall=result.recall,
-                                                         fvalue=result.fvalue)
-        except Exception as e:
-            self.logger.error(str(e))
-            self.logger.error(traceback.format_exc())
-            return drucker_pb2.EvaluateModelResponse(num=0,
-                                                     accuracy=0,
-                                                     precision=0,
-                                                     recall=0,
-                                                     fvalue=0)
+        first_req = next(request_iterator)
+        save_path = first_req.data_path
+
+        test_data = b''.join([first_req.data] + [r.data for r in request_iterator])
+        result, details = self.app.evaluate(test_data)
+        metrics = drucker_pb2.EvaluationMetrics(num=result.num,
+                                                accuracy=result.accuracy,
+                                                precision=result.precision,
+                                                recall=result.recall,
+                                                fvalue=result.fvalue,
+                                                option=result.option)
+        tmp_path = self.app.get_eval_path(uuid.uuid4().hex)
```

Comment:
Why do you need to save the file to tmp_path?

Reply:
right... I just copied code in upload_model, but saved evaluation data never be chunk. I fixed it.

Reply:
You mean code in UploadModel?

```
        save_path = None
        tmp_path = self.app.get_model_path(uuid.uuid4().hex)
        Path(tmp_path).parent.mkdir(parents=True, exist_ok=True)
        with open(tmp_path, 'wb') as f:
            for request in request_iterator:
                save_path = request.path
                model_data = request.data
                f.write(model_data)
            f.close()
```

Here, value of `save_path` is assigned in `with open(..)` scope, so I think it is a good way to use tmp_file

Reply:
I remember that the reason why I designed this way is to avoid storing the data on memory. And another reason is that we need to guarantee that the model data is completely downloaded. 

So `Evaluate` method also need to be changed like this in the future. One solution is saving it to the tempfile like this. Or another solution is send 1 line (e.g. `readline` on File method) data and evaluate it sequentially.

Reply:
I got it! 

> So Evaluate method also need to be changed like this in the future

I also agree with this